diff --git a/Gemfile.lock b/Gemfile.lock
index 27a7e16..e02f6df 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -1,6 +1,6 @@
 GIT
   remote: git://github.com/redbadger/rugged.git
-  revision: c405e9f6b7d23300e0ae6b2175930920c743fa6d
+  revision: 6a1085acc4716c31f3fb354d88fab7157f65d3ba
   branch: backends
   submodules: true
   specs:
@@ -9,13 +9,13 @@ GIT
 PATH
   remote: .
   specs:
-    colonel (0.7.1)
-      elasticsearch (~> 1.0.6)
+    colonel (0.6.1)
+      elasticsearch (~> 1.0.0)
 
 GEM
   remote: https://rubygems.org/
   specs:
-    activesupport (4.2.5)
+    activesupport (4.2.0)
       i18n (~> 0.7)
       json (~> 1.7, >= 1.7.7)
       minitest (~> 5.1)
@@ -23,36 +23,34 @@ GEM
       tzinfo (~> 1.1)
     builder (3.2.2)
     coderay (1.1.0)
-    cucumber (2.1.0)
+    cucumber (1.3.18)
       builder (>= 2.1.2)
-      cucumber-core (~> 1.3.0)
       diff-lcs (>= 1.1.3)
-      gherkin3 (~> 3.1.0)
+      gherkin (~> 2.12)
       multi_json (>= 1.7.5, < 2.0)
-      multi_test (>= 0.1.2)
-    cucumber-core (1.3.0)
-      gherkin3 (~> 3.1.0)
+      multi_test (>= 0.1.1)
     diff-lcs (1.2.5)
     docile (1.1.5)
-    elasticsearch (1.0.15)
-      elasticsearch-api (= 1.0.15)
-      elasticsearch-transport (= 1.0.15)
-    elasticsearch-api (1.0.15)
+    elasticsearch (1.0.6)
+      elasticsearch-api (= 1.0.6)
+      elasticsearch-transport (= 1.0.6)
+    elasticsearch-api (1.0.6)
       multi_json
-    elasticsearch-transport (1.0.15)
+    elasticsearch-transport (1.0.6)
       faraday
       multi_json
-    faraday (0.9.2)
+    faraday (0.9.1)
       multipart-post (>= 1.2, < 3)
-    gherkin3 (3.1.2)
+    gherkin (2.12.2)
+      multi_json (~> 1.3)
     i18n (0.7.0)
-    json (1.8.3)
+    json (1.8.1)
     method_source (0.8.2)
-    minitest (5.8.3)
-    multi_json (1.11.2)
-    multi_test (0.1.2)
+    minitest (5.5.0)
+    multi_json (1.10.1)
+    multi_test (0.1.1)
     multipart-post (2.0.0)
-    pry (0.10.3)
+    pry (0.10.1)
       coderay (~> 1.1.0)
       method_source (~> 0.8.1)
       slop (~> 3.4)
@@ -71,16 +69,16 @@ GEM
     rspec-mocks (3.0.4)
       rspec-support (~> 3.0.0)
     rspec-support (3.0.4)
-    shoulda-matchers (3.0.1)
-      activesupport (>= 4.0.0)
-    simplecov (0.11.1)
+    shoulda-matchers (2.7.0)
+      activesupport (>= 3.0.0)
+    simplecov (0.9.1)
       docile (~> 1.1.0)
-      json (~> 1.8)
-      simplecov-html (~> 0.10.0)
-    simplecov-html (0.10.0)
+      multi_json (~> 1.0)
+      simplecov-html (~> 0.8.0)
+    simplecov-html (0.8.0)
     slop (3.6.0)
     thor (0.19.1)
-    thread_safe (0.3.5)
+    thread_safe (0.3.4)
     tzinfo (1.2.2)
       thread_safe (~> 0.1)
 
@@ -99,6 +97,3 @@ DEPENDENCIES
   shoulda-matchers
   simplecov
   thor
-
-BUNDLED WITH
-   1.10.6
diff --git a/README.md b/README.md
index 243c7c3..7dbbd04 100644
--- a/README.md
+++ b/README.md
@@ -19,8 +19,6 @@ And then execute:
 
 ### Dependencies
 
-Up to gem version 0.7.x the Colonel can work with elasticsearch 2.x version.
-
 The Colonel requires at least [elasticsearch](http://www.elasticsearch.org) 1.0 to work.
 
 NOTE: The Colonel currently doesn't work with elasticsearch 1.2 or later, which is a known issue
diff --git a/colonel.gemspec b/colonel.gemspec
index 198fd8b..992d327 100644
--- a/colonel.gemspec
+++ b/colonel.gemspec
@@ -16,7 +16,7 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
 
-  spec.add_runtime_dependency "elasticsearch", "~> 1.0.15"
+  spec.add_runtime_dependency "elasticsearch", "~> 1.0.0"
 
   spec.add_development_dependency "bundler", "~> 1.5"
   spec.add_development_dependency "thor"
diff --git a/lib/colonel.rb b/lib/colonel.rb
index febc228..361801f 100644
--- a/lib/colonel.rb
+++ b/lib/colonel.rb
@@ -21,25 +21,13 @@ require 'colonel/indexer'
 module Colonel
   # Public: Sets configuration options.
   #
-  # Colonel.config.storage_path               - location to store git repo on disk
-  # Colonel.config.index_name                 - the name of elasticsearch index to store into
-  # Colonel.config.elasticsearch_uri          - host for elasticsearch
-  # Colonel.config.rugged_backend             - storage backend, an instance of Rugged::Backend
-  # Colonel.config.elasticsearch_timeout_secs - request timeout for elasticsearch
+  # Colonel.config.storage_path       - location to store git repo on disk
+  # Colonel.config.index_name         - the name of elasticsearch index to store into
+  # Colonel.config.elasticsearch_uri  - host for elasticsearch
+  # Colonel.config.rugged_backend     - storage backend, an instance of Rugged::Backend
   #
   # Returns a config struct
   def self.config
-    defaults = {
-      :storage_path => 'storage',
-      :index_name => 'colonel-storage',
-      :elasticsearch_uri => 'localhost:9200',
-      :rugged_backend => nil,
-      :elasticsearch_timeout_secs => 60
-    }
-
-    ordered_struct_fields = defaults.keys
-    ordered_struct_values = ordered_struct_fields.map{|k| defaults[k]}
-
-    @config ||= Struct.new(*ordered_struct_fields).new(*ordered_struct_values)
+    @config ||= Struct.new(:storage_path, :index_name, :elasticsearch_uri, :rugged_backend).new('storage', 'colonel-storage', 'localhost:9200', nil)
   end
 end
diff --git a/lib/colonel/search/elasticsearch_provider.rb b/lib/colonel/search/elasticsearch_provider.rb
index 41cc898..8b7173a 100644
--- a/lib/colonel/search/elasticsearch_provider.rb
+++ b/lib/colonel/search/elasticsearch_provider.rb
@@ -155,19 +155,7 @@ module Colonel
 
       # Public: The Elasticsearch client
       def es_client
-        client_opts = {
-          host:            Colonel.config.elasticsearch_uri,
-          log:             false
-        }
-
-        if timeout = Colonel.config.elasticsearch_timeout_secs
-          if timeout <= 0
-            fail "Invalid Colonel.config.elasticsearch_timeout_secs: #{timeout} must be > 0"
-          end
-          client_opts[:request_timeout] = timeout
-        end
-
-        @es_client ||= ::Elasticsearch::Client.new(client_opts)
+        @es_client ||= ::Elasticsearch::Client.new(host: Colonel.config.elasticsearch_uri, log: false)
       end
 
       # Internal: idempotently create the ES index
@@ -195,8 +183,8 @@ module Colonel
           revision_type_name=> revision_mapping
         }
 
-        es_client.indices.put_mapping index: index_name, type: revision_type_name, body: revision_body
         es_client.indices.put_mapping index: index_name, type: type_name, body: item_body
+        es_client.indices.put_mapping index: index_name, type: revision_type_name, body: revision_body
 
         scopes.each do |name, preds|
           name = "#{type_name}_#{name}"
@@ -242,12 +230,10 @@ module Colonel
             # _id is "{id}-{state}"
             id: {
               type: 'string',
-              store: 'yes',
               index: 'not_analyzed'
             },
             state: {
               type: 'string',
-              store: 'yes',
               index: 'not_analyzed'
             },
             updated_at: {
diff --git a/lib/colonel/search/elasticsearch_result_set.rb b/lib/colonel/search/elasticsearch_result_set.rb
index ccd263b..9fea87c 100644
--- a/lib/colonel/search/elasticsearch_result_set.rb
+++ b/lib/colonel/search/elasticsearch_result_set.rb
@@ -9,7 +9,7 @@ module Colonel
   class ElasticsearchResultSet
     include Enumerable
 
-    attr_reader :total, :facets, :aggs
+    attr_reader :total, :facets
 
     # Internal: Create a new result set
     def initialize(results, document_type)
@@ -20,7 +20,6 @@ module Colonel
       @hits   = results["hits"]["hits"]
 
       @facets = results["facets"]
-      @aggs = results["aggregations"]
     end
 
     # Public: Iterate over raw results
diff --git a/lib/colonel/version.rb b/lib/colonel/version.rb
index ff2e33d..64a920c 100644
--- a/lib/colonel/version.rb
+++ b/lib/colonel/version.rb
@@ -1,3 +1,3 @@
 module Colonel
-  VERSION = "0.8.0"
+  VERSION = "0.6.1"
 end
diff --git a/provisioning/roles/elasticsearch/defaults/main.yml b/provisioning/roles/elasticsearch/defaults/main.yml
index 07d566c..5e7d82e 100644
--- a/provisioning/roles/elasticsearch/defaults/main.yml
+++ b/provisioning/roles/elasticsearch/defaults/main.yml
@@ -1,7 +1,7 @@
 %YAML 1.2
 ---
 elasticsearch:
-  version: 2.1.0
+  version: 1.0.0
   environment:
     - variable: ES_MIN_MEM
       value: 750m
